src/components/Editor/CallSites.js

```diff
@@ -44,20 +56,59 @@ class CallSites extends Component {
     selectedLocation: Object
   };
 
+  constructor() {
+    super();
+
+    this.state = {
+      topVisibleLine: 0,
+      bottomVisibleLine: 20,
+      leftVisibleColumn: 0,
+      rightVisibleColumn: 300
+    };
+  }
+
   componentDidMount() {
     const { editor } = this.props;
     const codeMirrorWrapper = editor.codeMirror.getWrapperElement();
 
     codeMirrorWrapper.addEventListener("click", e => this.onTokenClick(e));
+    editor.codeMirror.on("scroll", this.onEditorScroll);
   }
 
   componentWillUnmount() {
     const { editor } = this.props;
     const codeMirrorWrapper = editor.codeMirror.getWrapperElement();
 
     codeMirrorWrapper.removeEventListener("click", e => this.onTokenClick(e));
+    editor.codeMirror.off("scroll", this.onEditorScroll);
   }
 
+  onEditorScroll = debounce(e => {
+    const { editor } = this.props;
+
+    //
+    const charWidth = editor.codeMirror.defaultCharWidth();
+    const scrollArea = editor.codeMirror.getScrollInfo();
+    const { scrollLeft } = editor.codeMirror.doc;
+
+    // Get scroll position
+    const rect = editor.codeMirror.getWrapperElement().getBoundingClientRect();
+    const topVisibleLine = editor.codeMirror.lineAtHeight(rect.top, "window");
+    const bottomVisibleLine = editor.codeMirror.lineAtHeight(
+      rect.bottom,
+      "window"
+    );
```

Comment:
can we move this into an editor helper that the current scroll logic could use too?

`utils/editor`